### PR TITLE
updated version of PR 440 to add spark parameter validation 

### PR DESCRIPTION
**Pull request by eordentlich**
rebased onto latest branch-23.10 changes and addressing comments
one issue is the setting of spark params based on cuml params.  However, all current such cases have compatible values or are removed from validation.

### CODE REVIEW

**Comment by eordentlich** [comment]
build

**Comment by eordentlich** [comment]
build

**Comment by eordentlich** [comment]
build

**Comment by eordentlich** [comment]
build

**Comment by eordentlich** [comment]
build

File: python/tests/test_logistic_regression.py [hunk 1]
```diff
@@ -657,7 +658,7 @@ def test_compat_multinomial(
                 regParam=0.1,
                 elasticNetParam=0.2,
                 fitIntercept=fit_intercept,
-                family="multimonial",
+                family="multinomial",
```
Comment (wbo4958) [inline python/tests/test_logistic_regression.py:661]: good catch

**Comment by wbo4958** [comment]
LGTM.

**Review by wbo4958 — APPROVED** [review]

**Pull request merged**